skrub/_table_vectorizer.py

```diff
@@ -474,6 +483,126 @@ def _clone_transformers(self):
+        """
+        Split univariate transformers into multiple transformers, one for each
+        column. This is useful to use the inherited `ColumnTransformer` class
+        parallelism.
+
+        Parameters
+        ----------
+        during_fit : bool, default=False
+            Whether the method is called during `fit_transform` (True) or
+            during `transform` (False). This is used to determine whether
+            to split the self.transformers_ attribute (when False) or the
+            self.transformers attribute (when True).
+        """
+        if during_fit:
+            new_transformers = []
+            for name, trans, cols in self.transformers:
+                if (
+                    (not type(trans) == str)
+                    and trans._get_tags().get("univariate", False)
+                    and len(cols) > 1
+                ):
+                    for i, col in enumerate(cols):
+                        new_transformers.append(
+                            (name + "_split_" + str(i), clone(trans), [col])
+                        )
+                else:
+                    new_transformers.append((name, trans, cols))
+            self.transformers = new_transformers
+        else:
+            check_is_fitted(self, attributes=["transformers_"])
+            self._transformers_fitted_original = self.transformers_
+            new_transformers_ = []
+            new_transformer_to_input_indices = {}
+            for name, trans, cols in self.transformers_:
+                if (
+                    (not type(trans) == str)
+                    and trans._get_tags().get("univariate", False)
+                    and len(cols) > 1
+                ):
+                    splitted_transformers_ = trans._split()
+                    for i, (col, trans, trans_to_mapping) in enumerate(
+                        zip(
+                            cols,
+                            splitted_transformers_,
+                            self._transformer_to_input_indices[name],
+                        )
+                    ):
+                        name_split = f"{name}_split_{i}"
+                        new_transformers_.append((name_split, trans, [col]))
+                        new_transformer_to_input_indices[name_split] = [
+                            trans_to_mapping
+                        ]
+                else:
+                    new_transformers_.append((name, trans, cols))
+                    new_transformer_to_input_indices[
+                        name
+                    ] = self._transformer_to_input_indices[name]
+            self.transformers_ = new_transformers_
+            self._transformer_to_input_indices = new_transformer_to_input_indices
+
+    def _merge_univariate_transformers(self):
+        # merge back self.transformers and self.transformers_
+        check_is_fitted(self, attributes=["transformers_"])
+        # merge self.transformers
+        new_transformers = []
```

Comment:
To improve readability, I suggest using two subfunctions outside of the scope of the class here. These subfunctions would respectively have `self.transformers` and `self.transformers_` for inputs. WDYT?